roxie/ccd/ccdserver.cpp

```diff
@@ -21364,7 +21364,17 @@ class CRoxieServerDiskReadBaseActivity : public CRoxieServerActivity, implements
                 varFileInfo.setown(resolveLFN(fileName, isOpt));
                 numParts = 0;
                 if (varFileInfo)
+                {
                     numParts = varFileInfo->getNumParts();
+                    const IPropertyTree *options =  varFileInfo->queryProperties();
+                    if (options)
+                    {
+                        size32_t dfsSize = options->getPropInt("@recordSize");
+                        CachedOutputMetaData diskMeta(helper.queryDiskRecordSize()->querySerializedDiskMeta());
```

Comment:
@richardkchapman it probably needs a test case.  E.g., create a grouped persist and then try and read that.